### ISSUE TITLE
Request For HD Collection Textures Or Re-Enhanced Textures.

### ISSUE THREAD

**Issue by BlazeTorch** [VFansss/mgs2-v-s-fix#137]
some visual enhancements would be great for this old game.
model enhancements would also be great i think.
I'm also interested in modding this game if possible.

**Comment by UltimateNova1203** [comment]
I have a repo open for this project already, however we have a hurdle to clear before we can even really bring the hi-res textures over. The engine has a lot of limitations inherited from the Xbox conversion. There is a limit to how many assets can be in active memory for each stage (everything is a stage, from the main menu, to the cutscenes, to the actual playable stages themselves), and the game streams in new assets as it needs them.

There is, from what I can see, about a 16MB limit on textures, audio and models. We can change the textures out with anything we want (to a degree, resolution has some nuance), however we quickly exceed that 16MB limit due to the DXT5 textures. It doesn't compress alpha that well, so we get very large texture sizes.

We can truncate textures in a similar way to how the original textures were done, so a 1024*2048 texture may only actually use 1024*1152 of that size, the rest being blank. We can truncate the remaining height (I believe it's intervals of 24px) to shave some size down, but to get anything really usable, we have to overcome the size limitations someway.

I can see using LUA to intercept the texture calls and storing the new textures elsewhere in RAM, kinda similar to some emulators with high res assets do. Another idea is to leave the game as it is for the most part, but where it stores the pointer for the texture, have that redirected to another point in RAM where our hi-res stuff is located at.

I'm not sure how familiar V is with doing stuff like this, or if nemesis2000 is even interested with it. It may be worth looking into how GTAIII/VC/SA handles their hi-res texture mods with their plugins, as the plugin system that is used here seems to be similar.

Tl;Dr, overcome the cap on how much all the assets can take in RAM, and it'll be relatively trivial to port or remake the textures in even higher quality (this is what is preventing me from finishing my repo).

**Comment by BlazeTorch** [comment]
interesting, thank you for the in-depth details! if it's possible to release enhancement mods that would be amazing.  
this game is long overdue for it. it's unfortunate the HD collection is not yet on pc. so enhancement mods would
be the next to best thing.  it being said if it can be done definitely please upload it!

**Comment by theDarkFrenchy** [comment]
"it's unfortunate the HD collection is not yet on pc. "
Well it is... if you check RPSC3 + a copy of the game. I can do this in my country, because it's tolerated to have a copy of your game, as long  as you have bought a physical copy of the game.  
by the way, what bout this project, here on Gihub : 
[https://github.com/UltimateNova1203/mgs2-pchd](url)

**Comment by BlazeTorch** [comment]
thank you. i tested the HD collection through emulation but it has too much slow frame rate on my setup.
for the moment the PC version of MGS2 plays better on my setup. until emulation performance is better improved.

**Comment by theDarkFrenchy** [comment]
ah .. ok. I don't use PS3 emulator. I saw the video on youtube and it's impressive !! 
back to the PC version !!  :-)
